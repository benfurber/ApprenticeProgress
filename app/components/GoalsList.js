// @flow

import React, { Component } from "react";
import { FlatList } from "react-native";
import { Content } from "native-base";

import { GoalSummary } from "components";
import type { GoalsDataType, navigationType } from "types";
import { orderByTitle } from "utils";

type Props = { goals: GoalsDataType, navigation: navigationType };

class GoalsList extends Component<Props> {
  render() {
    let { goals } = this.props;
    goals = orderByTitle(goals);

    return (
      <Content>
        <FlatList
          data={goals}
          renderItem={({item}) => this._renderGoal(item)}
          keyExtractor={item => item.id.toString()}
        />
      </Content>
    );
  };

  _renderGoal(goal) {
    return (
      <GoalSummary
        details={goal}
        navigation={this.props.navigation}
      />
    );
  }
};

export { GoalsList };
