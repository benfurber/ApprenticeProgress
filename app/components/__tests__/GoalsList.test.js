// @flow

import React from "react";
import { shallow } from "enzyme";

import { GoalsList } from "../GoalsList";
import { GoalSummary } from "../GoalSummary";

describe("GoalsList", () => {
  it("has a valid snapshot", () => {
    const goals = {
      [1]: {
        description: "Hello world 1",
        title: "Testing title",
        tag: "Tag",
      },
    };
    const component = shallow(<GoalsList goals={goals} />);

    expect(component).toMatchSnapshot();
  });

  it("renders the right number of GoalSummaries", () => {
    const goals = {
      [1]: {
        description: "Hello world 1",
        title: "Testing title",
        tag: "Tag",
      },
      [2]: {
        description: "Hello world 2",
        title: "Testing title",
        tag: "Tag",
      },
    };
    const component = shallow(<GoalsList goals={goals} />);

    expect(component.find(GoalSummary)).toHaveLength(2);
  });
});
