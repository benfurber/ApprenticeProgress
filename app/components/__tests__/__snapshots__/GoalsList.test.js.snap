// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalsList has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoalsList
    goals={
      Array [
        Object {
          "description": "Hello world 1",
          "id": 1,
          "tag": "I'm team thoughtbot",
          "title": "Testing title",
        },
      ]
    }
    navigation={
      Object {
        "getParam": [Function],
        "navigate": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <FlatList
        data={
          Array [
            Object {
              "description": "Hello world 1",
              "id": 1,
              "tag": "I'm team thoughtbot",
              "title": "Testing title",
            },
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": Array [
          Object {
            "description": "Hello world 1",
            "id": 1,
            "tag": "I'm team thoughtbot",
            "title": "Testing title",
          },
        ],
        "disableVirtualization": false,
        "horizontal": false,
        "initialNumToRender": 10,
        "keyExtractor": [Function],
        "maxToRenderPerBatch": 10,
        "numColumns": 1,
        "onEndReachedThreshold": 2,
        "renderItem": [Function],
        "scrollEventThrottle": 50,
        "updateCellsBatchingPeriod": 50,
        "windowSize": 21,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <FlatList
          data={
            Array [
              Object {
                "description": "Hello world 1",
                "id": 1,
                "tag": "I'm team thoughtbot",
                "title": "Testing title",
              },
            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "description": "Hello world 1",
              "id": 1,
              "tag": "I'm team thoughtbot",
              "title": "Testing title",
            },
          ],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
